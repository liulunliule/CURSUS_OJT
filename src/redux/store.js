import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
