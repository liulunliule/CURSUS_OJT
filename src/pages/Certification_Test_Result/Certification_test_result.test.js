import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Certification_test_result from '.';

test('test Certification Test Result Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
        <MemoryRouter>
            <Certification_test_result/>
        </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
