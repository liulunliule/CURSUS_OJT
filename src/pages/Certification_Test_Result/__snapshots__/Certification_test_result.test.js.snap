// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Certification Test Result Page 1`] = `
<div
  className="test_result"
>
  <div
    className="toolbar_certification"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12"
        >
          <div
            className="toolbar_title"
          >
            <div
              className="titleleft"
            >
              <div
                className="toolbar_title_item"
              >
                <nav
                  aria-label="breadcrumb"
                >
                  <ol
                    className="breadcrumb"
                  >
                    <li
                      className="breadcrumb-item"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        Home
                      </a>
                    </li>
                    <li
                      className="breadcrumb-item"
                    >
                      <a
                        href="/secondLayout/certification_center"
                        onClick={[Function]}
                      >
                        Certification Center
                      </a>
                    </li>
                    <li
                      aria-current="page"
                      className="breadcrumb-item active"
                    >
                      Result
                       
                    </li>
                  </ol>
                </nav>
              </div>
            </div>
            <div
              className="titleright"
            >
              <a
                className="blog_link"
                href="/secondLayout/certification_center"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-angles-left "
                  data-icon="angles-left"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160zm352-160l-160 160c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L301.3 256 438.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                 Back to Certification Center
              </a>
            </div>
          </div>
          <div
            className="main_title"
          >
            <h2>
              Result
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="row justify-content-md-center"
    >
      <div
        className="col-md-6"
      >
        <div
          className="certi_form stickytime"
        >
          <div
            className="test_result_bg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
